src/lib/logger.ts

```diff
@@ -106,6 +113,39 @@ export class Logger {
     Logger.log(event, parameters);
   }
 
+  public static logToolChange(change: ToolChangeEventType, toolId: string) {
+    let parameters: {[k: string]: any} = {
+      toolId,
+      ...change
+    };
+
+    // either flatten change.properties (for graphs) or change.data (for drawing tool),
+    // unless properties/data an array, in which case standardize on "properties" and clean up.
```

Comment:
I'm inclined to think that we should define the format that the logger expects and it's up to the tools to convert/conform their change object to the logger's expectations. It shouldn't be the logger's job to know about the implementation details of the other tools.

Reply:
I partly agree, and added a couple change.

The log function now requires the `operation` name string, and both the geometry and drawing tools pre-flatten the change data, so that the logger doesn't need to know about the details of the change object. The logger now blindly logs whatever is passed to it.

However, this does not enforce a single structure for the log data, as I think it important for all the events to be logged with all their relevant data, and the smallest union of all the different events' data is simply the union of all the change events, which is what we are already requiring.